src/ui/pages/AssetInfo/AssetResources.tsx

```diff
@@ -0,0 +1,365 @@
+        aria-label="Copy token address"
+        size={36}
+        kind="neutral"
+        className="parent-hover"
+        style={{
+          ['--parent-content-color' as string]: 'var(--neutral-400)',
+          ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+          ['--button-background' as string]: 'var(--neutral-200)',
+          ['--button-background-hover' as string]: 'var(--neutral-300)',
+          paddingInline: 8,
+        }}
+      >
+        <HStack gap={4} alignItems="center">
+          <HStack gap={6} alignItems="center">
+            <img
+              src={network.icon_url}
+              alt={network.name}
+              width={24}
+              height={24}
+            />
+            <UIText kind="small/accent">Explorer</UIText>
+          </HStack>
+          <LinkIcon
+            className="content-hover"
+            style={{ width: 20, height: 20 }}
+          />
+        </HStack>
+      </Button>
+    );
+  }
+  return (
+    <Button
+      onClick={handleCopy}
+      aria-label="Copy token address"
+      size={36}
+      kind="neutral"
+      className="parent-hover"
+      style={{
+        paddingInline: 8,
+        ['--parent-content-color' as string]: 'var(--neutral-400)',
+        ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+        ['--button-background' as string]: 'var(--neutral-200)',
+        ['--button-background-hover' as string]: 'var(--neutral-300)',
+      }}
+    >
+      <HStack gap={4} alignItems="center">
+        <HStack gap={6} alignItems="center">
+          <img
+            src={network.icon_url}
+            alt={network.name}
+            width={24}
+            height={24}
+          />
+          <UIText kind="small/accent">
+            {middleTruncate({
+              value: address,
+              leadingLettersCount: 5,
+              trailingLettersCount: 4,
+            })}
+          </UIText>
+        </HStack>
+        {isSuccess ? (
+          <CheckIcon
+            style={{ color: 'var(--positive-500)', width: 20, height: 20 }}
+          />
+        ) : (
+          <CopyIcon
+            className="content-hover"
+            style={{ width: 20, height: 20 }}
+          />
+        )}
+      </HStack>
+    </Button>
+  );
+}
+
+function CopyAddressButton({ address }: { address: string }) {
+  const { handleCopy, isSuccess } = useCopyToClipboard({ text: address });
+
+  return (
+    <UnstyledButton
+      onClick={handleCopy}
+      aria-label="Copy token address"
+      className="parent-hover"
+      style={{
+        ['--parent-content-color' as string]: 'var(--neutral-400)',
+        ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+        display: 'flex',
+      }}
+    >
+      {isSuccess ? (
+        <CheckIcon style={{ color: 'var(--positive-500)' }} />
+      ) : (
+        <CopyIcon className="content-hover" />
+      )}
+    </UnstyledButton>
+  );
+}
+
+function AssetImplementationsDialog({
```

Comment:
```suggestion
function AssetImplementationsDialogContent({
```
Perhaps this is a better name? Since this component doesn't render the actual dialog component